### PR TITLE
[AF-1555]: Upgrade to Wildfly14

### PR DESCRIPTION
**Pull request by adrielparedes**
## Description

Upgrate to Wildfly 14

## Related to:
* https://github.com/jboss-integration/jboss-integration-platform-bom/pull/427
* https://github.com/errai/errai/pull/363
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/843
* https://github.com/kiegroup/appformer/pull/483
* https://github.com/kiegroup/kie-wb-common/pull/2147
* https://github.com/kiegroup/jbpm/pull/1336
* https://github.com/kiegroup/jbpm-wb/pull/1206
* https://github.com/kiegroup/drools/pull/2086
* https://github.com/kiegroup/droolsjbpm-integration/pull/1583
* https://github.com/kiegroup/kie-wb-distributions/pull/821 
* https://github.com/kiegroup/kie-docs/pull/1108
* https://github.com/kiegroup/optaplanner/pull/437
* https://github.com/kiegroup/optaplanner-wb/pull/304

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

File: kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-distribution/pom.xml [hunk 1]
```diff
@@ -315,7 +315,7 @@
                 <artifactItem>
                   <groupId>org.wildfly</groupId>
                   <artifactId>wildfly-dist</artifactId>
-                  <version>11.0.0.Final</version>
+                  <version>14.0.1.Final</version>
```
[the remark this answers is not in the document]
Comment (jhrcek) [reply kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-distribution/pom.xml]: @adrielparedes this comment still applies.

**Pull request merged**